hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RegionCoprocessorHost.java

```diff
@@ -275,11 +280,11 @@ public RegionCoprocessorHost(final HRegion region,
     out: for (RegionCoprocessorEnvironment env: coprocEnvironments) {
       if (env.getInstance() instanceof RegionObserver) {
         Class<?> clazz = env.getInstance().getClass();
-        for(;;) {
-          if (clazz == null) {
-            // we must have directly implemented RegionObserver
-            hasCustomPostScannerFilterRow = true;
-            break out;
+        while (clazz != null) {
+          if (clazz == Object.class) {
+            // we dont need to look postScannerFilterRow into Object class
+            clazz = null;
+            continue;
```

Comment:
U can basically break at this point? Come out of the inner loop.  
Could have kept for(;;) {  as is also.
Just to keep the change to be 3 lines only
Minor suggestion though